docs/reference/docs/concurrency-control.asciidoc

```diff
@@ -0,0 +1,111 @@
+[[optimistic-concurrency-control]]
+== Optimistic concurrency control
+
+Elasticsearch is distributed. When documents are created, updated, or deleted,
+the new version of the document has to be replicated to other nodes in the cluster. 
+Elasticsearch is also asynchronous and concurrent, meaning that these replication
+requests are sent in parallel, and may arrive at their destination out of sequence. 
+Elasticsearch needs a way of ensuring that an older version of a document never 
+overwrites a newer version.
+
+To ensure this, every operation performed to a document is assigned a sequence number
+by the primary shard that coordinates that change. The sequence number is increased
+with each operation and thus newer operations are guaranteed to have a higher sequence
+number. Elasticsearch can then use the sequence number of operations to make sure they
+never override a newer document version is never overridden by a change that has a 
+smaller sequence number assigned to it. 
+
+For example, the following indexing command will create a document and assign it its 
+initial sequence number and primary term:
+
+[source,js]
+--------------------------------------------------
+PUT products/_doc/1567
+{
+    "product" : "r2d2",
+    "details" : "A resourceful astromech droid"
+}
+--------------------------------------------------
+// CONSOLE
+
+You can see the assigned sequence number and primary term in the 
+the `_seq_no` and `_primary_term` fields of the response:
+
+[source,js]
+--------------------------------------------------
+{
+    "_shards" : {
+        "total" : 2,
+        "failed" : 0,
+        "successful" : 1
+    },
+    "_index" : "products",
+    "_type" : "_doc",
+    "_id" : "1567",
+    "_version" : 1,
+    "_seq_no" : 362,
+    "_primary_term" : 2,
+    "result" : "created"
+}
+--------------------------------------------------
+// TESTRESPONSE[s/"_seq_no" : \d+/"_seq_no" : $body._seq_no/ s/"_primary_term" : 2/"_primary_term" : $body._primary_term/]
+
+
+Elasticsearch keeps tracks of the sequence number and primary of the last
+operation to have changed each of the document it stores. These are returned
```

Comment:
Elasticsearch keeps tracks of the sequence number and primary term for every document it stores, this value will change based on the last operation to modify the document. The `_seq_no` and `_primary_term` fields are returned

Reply:
I will adopt the second part of the suggestion, but I feel the first part is not accurate - there's no sequence number of a document but rather a seq of the last operation to have changed the doc.